docs: Getting started sandbox (#4045)
grafana/tempo: docs/sources/tempo/getting-started/docker-example.md

menuTitle: Quick start for Tempo
description: Use Docker to quickly view traces using K-6 and Tempo
weight: 600
killercoda:
  title: Quick start for Tempo
  description: Use Docker to quickly view traces using K-6 and Tempo
  details:
      intro:
         foreground: docker-compose-update.sh

>     backend:

Reviewer: ```suggestion
  backend:
    backend:
```

menuTitle: Quick start for Tempo
description: Use Docker to quickly view traces using K-6 and Tempo
weight: 600
killercoda:
  title: Quick start for Tempo
  description: Use Docker to quickly view traces using K-6 and Tempo
  details:
      intro:
         foreground: docker-compose-update.sh
  preprocessing:
    substitutions:
      - regexp: ../../
        replacement: /docs/tempo/latest/

>       - regexp: '{{< relref "./instrumentation" >}}'

Reviewer: @jdbaldry I know the Witers' Toolkit recommends using full URLs. Should I update the URls to all use the full URLs or does it matter that much?

Reviewer 2: Using full URLs would let us remove these preprocessing statements which would be nice. The transformer tool knows how to convert full URLs with version substitution into Killercoda URLs.

Reviewer: Okay. If I change the URLs in the body, then the regexp can be removed?

Reviewer: Okay. I'll remove the preprocess notes from the frontmatter.

The Tempo examples running with Docker using docker-compose include a version of Tempo and a storage configuration suitable for testing or development.

<!-- INTERACTIVE ignore START -->

## Before you begin

To follow this guide, you need:

- Git
- [Docker](https://docs.docker.com/compose/install/)
- The Docker Compose plugin (included with Docker Desktop)

{{< admonition type="tip" >}}
Alternatively, you can try out this example in our interactive learning environment: [Quick start for Tempo](https://killercoda.com/grafana-labs/course/tempo/quick-start).

It's a fully configured environment with all the dependencies installed.

> 
> 
> Provide feedback, report bugs, and raise issues in the [Grafana Killercoda repository](https://github.com/grafana/killercoda).

Reviewer 3: ```suggestion

Provide feedback, report bugs, and raise issues in the [Grafana Killercoda repository](https://github.com/grafana/killercoda).
```

Extra CR/LF

Alloy includes a banner image at this point. There's no banner here?

Also, this is new compared to the Alloy version. Should I include this over in the Alloy tutorials?

Reviewer: @Jayclifford345 What do you think?

Author: Yeah, more than happy to have the banner removed and this section updated. I still think some form of icon or branding, later on, to differentiate it from a normal admonition makes users know this tutorial includes a sandbox from the initial glance.

This quick start guide uses the `local` example running Tempo as a single binary (monolithic). Any data is stored locally in the `tempo-data` folder.
To learn more, read the [local storage example README](https://github.com/grafana/tempo/blob/main/example/docker-compose/local).

> ## Alternative: Complete MLTP example

Reviewer: Why did you remove this section?

- Docker
- The docker-compose plugin

{{< admonition type="tip" >}}
Alternatively, you can try out this example in our interactive learning environment: [Quick start for Tempo](https://killercoda.com/grafana-labs/course/tempo/quick-start).

It's a fully configured environment with all the dependencies already installed.

![Interactive](/media/docs/tempo/tempo-ile.svg)

Provide feedback, report bugs, and raise issues in the [Grafana Killercoda repository](https://github.com/grafana/killercoda).
{{< /admonition >}}

<!-- INTERACTIVE ignore END -->

> {{<docs/ignore>}}

Reviewer: Normally we use and admonition shortcode. Do we need to remove any shortcodes for Killercode?

> **Note**: A setup script is running in the background to install the necessary dependencies. This should take no longer than 30 seconds. Your instance will be ready to use once you `Setup complete. You may now begin the tutorial`.

> {{</docs/ignore>}}

Reviewer: Normally we use and admonition shortcode. Do we need to remove any shortcodes when we use markdown for Killercoda and Hugo?

Reviewer 2: We have support for some shortcodes in Killercoda but not all of them. Admonitions are not yet supported so we write Killercoda notes in plain Markdown. I've created the documentation in https://github.com/grafana/killercoda/pull/118.

Reviewer 2: https://github.com/grafana/killercoda/pull/121 🎉

Author: tested and works perfectly: https://killercoda.com/grafana-dev-testing/course/tempo/quick-start

- Docker
- The docker-compose plugin

{{< admonition type="tip" >}}
Alternatively, you can try out this example in our interactive learning environment: [Quick start for Tempo](https://killercoda.com/grafana-labs/course/tempo/quick-start).

It's a fully configured environment with all the dependencies already installed.

> ![Interactive](/media/docs/tempo/tempo-ile.svg)

Reviewer: This large banner is out of place in the docs. The admonition already highlights the tutorial. Isn't the text in the green highlight enough? 
```suggestion
```
This is what it looks like: 
![image](https://github.com/user-attachments/assets/ecc68bed-7dfc-45be-89ac-147909e3dc7c)

> ![Interactive](/media/docs/tempo/tempo-ile.svg)

Author: Hi @knylander-grafana, 
Not a problem we can get that removed. We have made it standard across the other pages. Agreed it might be to big. I can remove for now and speak to the design team to see if we can make something less aggressive but still indicate that is page is part of the greater interactive sandbox effort

<!-- INTERACTIVE page step1.md END -->

<!-- INTERACTIVE page step2.md START -->

## Explore the traces in Grafana

As part of the docker compose manifest, Grafana is now accessible on port 3000. We can use Grafana to explore the traces generated by the k6-tracing service.

1. Open a browser and navigate to [http://localhost:3000](http://localhost:3000).

1. Once logged in, navigate to the **Explore** page, select the **Tempo** data source and select the **Search** tab. Select **Run query** to list the recent traces stored in Tempo. Select one to view the trace diagram:
   
   {{< figure align="center" src="../../setup/assets/setup-linux-run-query.png" alt="Query example" >}}
    

2. A couple of minutes after Tempo has started up successfully, select the **Service graph** tab for the Tempo data source in the **Explore** page. Select **Run query** to view a service graph, generated by Tempo’s metrics-generator.

   {{< figure align="center" src="../../setup/assets/setup-linux-node-graph.png" alt="Service graph sample" >}}

3. To stop the services:
    ```bash
    docker compose down -v
    ```
    
<!-- INTERACTIVE page step2.md END -->

<!-- INTERACTIVE page finish.md START -->

> # Next steps

Reviewer: This is an automated thing from the validator. I'm not sure how to escape around this. 

@jdbaldry Have you dealt with this before?

> # Next steps

Reviewer 2: Ooh this is a good one. This is usually a correct positive because a page shouldn't have two H1 headings but it makes sense use a H1 here because it's the first heading in that course's page.

I think `doc-validator` needs a comment based mechanism for ignoring parts of the code and I may be able to find some time to work on that today.

In the short term, you can make this section start with an H2 and know that the transformer tool will make it an H1: https://github.com/grafana/killercoda/blob/staging/tools/transformer/transform.go#L233-L262

> # Next steps

Author: Hi all, sorry that was completely my mistake. It should have been a H2 so now made that fix!

   local-tempo-1        "/tempo -config.file…"   tempo               running             0.0.0.0:3200->3200/tcp, 0.0.0.0:4317-4318->4317-4318/tcp, 0.0.0.0:9411->9411/tcp, 0.0.0.0:14268->14268/tcp
   ```

<!-- INTERACTIVE page step1.md END -->

<!-- INTERACTIVE page step2.md START -->

## Explore the traces in Grafana

> As part of the docker compose manifest, Grafana is now accessible on port 3000. 

Reviewer 3: ```suggestion
As part of the Docker Compose manifest, Grafana is now accessible on port 3000. 
```

According to:  https://docs.docker.com/compose/  it should be initial caps when describing the command.

Reviewer: Good catch. I found this earlier and corrected it but missed this occurrence.

1. Open a browser and navigate to [http://localhost:3000](http://localhost:3000).

1. Once logged in, navigate to the **Explore** page, select the **Tempo** data source and select the **Search** tab. Select **Run query** to list the recent traces stored in Tempo. Select one to view the trace diagram:
   
   {{< figure align="center" src="/media/docs/grafana/data-sources/tempo/query-editor/tempo-ds-builder-span-details-v11.png" alt="Use the query builder to explore tracing data in Grafana" >}}
    

1. A couple of minutes after Tempo starts, select the **Service graph** tab for the Tempo data source in the **Explore** page. Select **Run query** to view a service graph, generated by Tempo’s metrics-generator.

   {{< figure align="center" src="/media/docs/grafana/data-sources/tempo/query-editor/tempo-ds-query-service-graph.png" alt="Service graph sample" >}}

1. To stop the services:
    ```bash
    docker compose down -v
    ```
    
<!-- INTERACTIVE page step2.md END -->

<!-- INTERACTIVE page finish.md START -->

## Next steps

You have successfully set up Tempo and Grafana to explore traces generated by the k6-tracing service.

### Alternative: Complete MLTP example

> If you would like to use a demo with Tempo, Mimir, Loki, and Grafana, then try the [Introduction to Metrics, Logs, Traces, and Profiling in Grafana](https://github.com/grafana/intro-to-mlt).

Reviewer: ```suggestion
If you would like to use a demo with multiple telemetry signals, then try the [Introduction to Metrics, Logs, Traces, and Profiling in Grafana](https://github.com/grafana/intro-to-mlt).
```

> `Intro-to-mltp` provides a self-contained environment for learning about Mimir, Loki, Tempo, and Grafana.

Reviewer: ```suggestion
`Intro-to-mltp` provides a self-contained environment for learning about Mimir, Loki, Tempo, Pyroscope, and Grafana.
```

The project includes detailed explanations of each component and annotated configurations for a single-instance deployment.
Data from `intro-to-mltp` can also be pushed to Grafana Cloud.

### Further reading
Here are some resources to help you learn more about Tempo:

> * [Instrumenting your application]({{< relref "./instrumentation" >}})
> * [Setup guides for Tempo]({{< relref "../setup" >}})

Reviewer: ```suggestion
* [Instrumenting your application](https://grafana.com/docs/tempo/<TEMPO_VERSION>/getting-started/instrumentation/)
* [Setup guides for Tempo](https://grafana.com/docs/tempo/<TEMPO_VERSION>/setup/)
```
